backend/src/layer/data/swka_parser/html_parser.rs

```diff
@@ -397,12 +400,28 @@ impl HTMLParser {
             .collect()
     }
 
-    fn get_dish_type(dish_node: &ElementRef) -> Option<FoodType> {
-        let dish_type_node = dish_node.select(&DISH_TYPE_NODE_CLASS_SELECTOR).next()?;
-        dish_type_node
-            .value()
-            .attr(DISH_TYPE_ATTRIBUTE_NAME)
-            .map(FoodType::parse)
+    fn get_dish_type(dish_node: &ElementRef, name: &str) -> Option<FoodType> {
+        let dish_type_node = dish_node.select(&DISH_TYPE_NODE_CLASS_SELECTOR).next();
+        if let Some(dish_type_node) = dish_type_node {
+            let preliminary_dish_type = dish_type_node
+                .value()
+                .attr(DISH_TYPE_ATTRIBUTE_NAME)
+                .map(FoodType::parse);
+            if preliminary_dish_type.is_none() || Some(FoodType::Unknown) == preliminary_dish_type {
+                return Self::check_for_poultry(name);
+            }
+            preliminary_dish_type
+        } else {
+            Self::check_for_poultry(name)
+        }
+    }
+
+    fn check_for_poultry(name: &str) -> Option<FoodType> {
```

Comment:
At least regarding to the function's name, I think returning a `bool` would be more idiomatic.